api/client/client.go

```diff
@@ -59,14 +63,16 @@ func NewServer(addr string) Remote {
 }
 
 func (srv *server) Hosts() []string {
-	return []string{net.JoinHostPort(srv.Address, fmt.Sprintf("%d", srv.Port))}
+	return []string{srv.URL}
 }
 
 func newServer(addr string) *server {
```

Comment:
`newServer` might make more sense as:
```Go
func newServer(u *url.URL) (*server, error) {
	[...]
}
```
This would avoid parsing the addr, converting it to string, reparsing it, reconverting it to string. And though this isn't your code originally, discarding an `error` value and returning `nil` is a bad practice in Go.